src/system_simulation/simulation_runner.rb

```diff
@@ -1,111 +1,160 @@
+# a modelica model of our geometry when necessary. This class provides public interfaces for different results of the
+# simulation.
 class SimulationRunner
+  include Singleton
 
-  def initialize(suppress_compilation=false, keep_temp_dir=false)
-    @model_name = "seesaw3"
+  def initialize(suppress_compilation = false, keep_temp_dir = false)
+    @model_name = 'seesaw3'
 
     if suppress_compilation
       @directory = File.dirname(__FILE__)
     else
       @directory = Dir.mktmpdir
       puts @directory
-      if not keep_temp_dir
-        ObjectSpace.define_finalizer(self, proc { FileUtils.remove_entry @directory })
-      end
+      ObjectSpace.define_finalizer(self, proc { FileUtils.remove_entry @directory }) unless keep_temp_dir
 
       run_compilation
     end
   end
 
-  def get_hub_time_series(hubIDs, stepSize, mass, constant=50)
+  def get_hub_time_series(hub_ids, step_size, mass, constant = 50)
     data = []
-    simulation_time = Benchmark.realtime { run_simulation(constant, mass, "node_pos.*") }
-    import_time = Benchmark.realtime { data = import_csv(File.join(@directory, "#{@model_name}_res.csv")) }
-    puts("simulation time: #{simulation_time.to_s}s csv parsing time: #{import_time.to_s}s")
+    simulation_time = Benchmark.realtime { run_simulation(constant, mass, 'node_pos.*') }
+    import_time = Benchmark.realtime { data = parse_data(read_csv) }
+    puts("simulation time: #{simulation_time}s csv parsing time: #{import_time}s")
     data
   end
 
-
-  def get_period(mass=20, constant=5000)
-    # TODO confirm correct result
-    run_simulation(constant, mass, "revLeft.phi")
+  def get_period(mass = 20, constant = 5000)
+    # TODO: confirm correct result
+    run_simulation(constant, mass, 'revLeft.phi')
 
     require 'gsl'
     require 'csv'
 
     stop_time = 10
 
-    data = CSV.read((File.join(@directory, "#{@model_name}_res.csv")), :headers=>true)['revLeft.phi']
-    vector = data.map{ |v| v.to_f }.to_gv
+    # TODO: make this call use read_csv
+    data = CSV.read(File.join(@directory, "#{@model_name}_res.csv"), headers: true)['revLeft.phi']
+    vector = data.map(&:to_f).to_gv
 
     sample_rate = vector.length / stop_time
 
     # https://github.com/SciRuby/rb-gsl/blob/master/examples/fft/fft.rb
     y2 = vector.fft.subvector(1, data.length - 2).to_complex2
     mag = y2.abs
-    f = GSL::Vector.linspace(0, sample_rate/2, mag.size)
-    #p mag.to_a
-    #p mag.max_index
-    #p f.to_a
-    return 1 / f[mag.max_index]
+    f = GSL::Vector.linspace(0, sample_rate / 2, mag.size)
+
+    1 / f[mag.max_index]
   end
 
-  def find_equilibrium(mass=20, constant=50)
-    # run simulation mit startTime 0.3; get first frame
-    run_simulation(constant, mass, "node_pos.*")
-    import_csv(File.join(@directory, "#{@model_name}_res.csv"))
+  # Returns index of animation frame when system is in equilibrium i.e. the mean of the angle difference
```

Comment:
I don't get the last part
> the mean of the angle difference
Maybe you can clarify, or just remove it

Reply:
Clarified in 6ee258a